src/app/shared/components/copy-button/copy-button.component.ts

```diff
@@ -9,12 +14,20 @@ export class CopyButtonComponent {
   public elRef = inject<ElementRef<HTMLElement>>(ElementRef<HTMLElement>);
   public copied = false;
 
+  public cdr = inject<ChangeDetectorRef>(ChangeDetectorRef);
+
   onCopy() {
     const preRef = this.elRef.nativeElement.querySelector('pre:not(.hide)');
     if (!preRef) {
       return;
     }
     navigator.clipboard.writeText(preRef.firstChild.textContent);
     this.copied = true;
+    this.cdr.detectChanges();
+
+    setTimeout(() => {
```

Comment:
```suggestion
    this.revertIconTimeout = setTimeout(() => {
```

Reply:
should we assign the result of the setTimeout call to a property and `clearTimeout` in the `onDestroy` lifecycle hook?